Readme updates

aaron-bray opened the pull request:
I rearranged the cmake section a bit, and added a big section to walk folks through cmake generation

I added a couple of blurbs for running a pipeline and pointing to maptk and viame as example users

I added a Code Structure and Provided Functionality section where we can talk about the cv algorithms/functionality kwiver provides (I do not feel qualified to describe these)
** Someone will need to fill these out **

mleotta commented on README.rst:
```diff
@@ -87,107 +87,162 @@ Fletch
-the code that depends on an external package is in a directory with
-the major dependency name (e.g. vxl, ocv). The dependencies can be
-turned ON or OFF through CMake variables.
+Note, This assumes your fletch was build with python support (Turn OFF if not)
```
typo "build" --> "built"

aaron-bray replied:
Ok, fixed

mleotta replied:
See other comment: fixed in the wrong spot

mleotta commented on README.rst:
```diff
@@ -87,107 +87,162 @@ Fletch
-.. _Boost: http://www.boost.org/
+You will also need to replace the fletch path with your own::
+
+cmake </path/to/kwiver/source> -Dfletch_DIR:PATH=<path/to/fletch/builds/install> -DCMAKE_BUILD_TYPE=Release -DKWIVER_ENABLE_ARROWS:BOOL=ON -DKWIVER_ENABLE_CERES:BOOL=ON -DKWIVER_ENABLE_C_BINDINGS:BOOL=ON -DKWIVER_ENABLE_LOG4CPLUS:BOOL=ON -DKWIVER_ENABLE_OPENCV:BOOL=ON -DKWIVER_ENABLE_PROCESSES:BOOL=ON -DKWIVER_ENABLE_PROJ:BOOL=ON -DKWIVER_ENABLE_PYTHON:BOOL=ON -DKWIVER_ENABLE_SPROKIT:BOOL=ON -DKW ...
```
You don't really need to specify everything here, just things differ from the default.   Because of a recent change, the specific arrows (OpenCV, VXL, etc.) are now enabled by default if you build the dependency if Fletch.  So I would suggest a more minimal subset:

    cmake </path/to/kwiver/source> -Dfletch_DIR:PATH=<path/to/fletch/builds/install>
                                   -DCMAKE_BUILD_TYPE=Release
                                   -DKWIVER_ENABLE_ARROWS:BOOL=ON
                                   -DKWIVER_ENABLE_C_BINDINGS:BOOL=ON
                                   -DKWIVER_ENABLE_PYTHON:BOOL=ON
                                   -DKWIVER_ENABLE_LOG4CPLUS:BOOL=ON
                                   -DKWIVER_ENABLE_SPROKIT:BOOL=ON
                                   -DKWIVER_ENABLE_PROCESSES:BOOL=ON
                                   -DKWIVER_ENABLE_TESTS:BOOL=OFF
                                   -DKWIVER_ENABLE_TOOLS:BOOL=ON
                                   -DKWIVER_USE_BUILD_TREE:BOOL=ON

aaron-bray commented:
ok, let me pull and refresh my build
I will just do what is different from default

On Fri, Jul 14, 2017 at 10:18 PM, Matt Leotta <notifications@github.com>
wrote:

> *@mleotta* commented on this pull request.
> ------------------------------
>
> In README.rst
> <https://github.com/Kitware/kwiver/pull/225#discussion_r127574234>:
>
> >
> -.. _Boost: http://www.boost.org/
> +You will also need to replace the fletch path with your own::
> +
> +cmake </path/to/kwiver/source> -Dfletch_DIR:PATH=<path/to/fletch/builds/install> -DCMAKE_BUILD_TYPE=Release -DKWIVER_ENABLE_ARROWS:BOOL=ON -DKWIVER_ENABLE_CERES:BOOL=ON -DKWIVER_ENABLE_C_BINDINGS:BOOL=ON -DKWIVER_ENABLE_LOG4CPLUS:BOOL=ON -DKWIVER_ENABLE_OPENCV:BOOL=ON -DKWIVER_ENABLE_PROCESSES:BOOL=ON -DKWIVER_ENABLE_PROJ:BOOL=ON -DKWIVER_ENABLE_PYTHON:BOOL=ON -DKWIVER_ENABLE_SPROKIT:BOOL=ON -DKWIVER_ENABLE_TESTS:BOOL=OFF -DKWIVER_ENABLE_TOOLS:BOOL=ON -DKWIVER_ENABLE_TRACK_ORACLE:BOOL=OFF -DKWIVER_ENABLE_VISCL:BOOL=OFF -DKWIVER_ENABLE_VXL:BOOL=ON -DKWIVER_USE_BUILD_TREE:BOOL=ON
>
> You don't really need to specify everything here, just things differ from
> the default. Because of a recent change, the specific arrows (OpenCV, VXL,
> etc.) are now enabled by default if you build the dependency if Fletch. So
> I would suggest a more minimal subset:
>
> cmake </path/to/kwiver/source> -Dfletch_DIR:PATH=<path/to/fletch/builds/install>
>                                -DCMAKE_BUILD_TYPE=Release
>                                -DKWIVER_ENABLE_ARROWS:BOOL=ON
>                                -DKWIVER_ENABLE_C_BINDINGS:BOOL=ON
>                                -DKWIVER_ENABLE_PYTHON:BOOL=ON
>                                -DKWIVER_ENABLE_LOG4CPLUS:BOOL=ON
>                                -DKWIVER_ENABLE_SPROKIT:BOOL=ON
>                                -DKWIVER_ENABLE_PROCESSES:BOOL=ON
>                                -DKWIVER_ENABLE_TESTS:BOOL=OFF
>                                -DKWIVER_ENABLE_TOOLS:BOOL=ON
>                                -DKWIVER_USE_BUILD_TREE:BOOL=ON
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Kitware/kwiver/pull/225#pullrequestreview-50179684>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AadLHzfuFtxlH0p8_KCKsXV3XWcJ-Rtsks5sOCF4gaJpZM4OYqbh>
> .
>

mleotta commented on README.rst:
```diff
@@ -87,107 +87,162 @@ Fletch
+Here are some applications using kwiver that serve as an example of how to leverage kwiver for a specific application
+
+======= ===================================================================
+MAP-Tk_  A collection of libraries and tools for making measurements from aerial video
```
I may update these descriptions

aaron-bray replied:
sounds good.
those are from the front of their readme files.

aaron-bray commented:
ok, i just took them from the front of their readme's

feel free to replace anything!

On Fri, Jul 14, 2017 at 10:20 PM, Matt Leotta <notifications@github.com>
wrote:

> *@mleotta* commented on this pull request.
> ------------------------------
>
> In README.rst
> <https://github.com/Kitware/kwiver/pull/225#discussion_r127574266>:
>
> >
> -Overview of Directories
> ------------------------
> +KWIVER Users
> +============
> +
> +Here are some applications using kwiver that serve as an example of how to leverage kwiver for a specific application
> +
> +======= ===================================================================
> +MAP-Tk_  A collection of libraries and tools for making measurements from aerial video
>
> I may update these descriptions
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Kitware/kwiver/pull/225#pullrequestreview-50179720>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AadLHwd_LC25EVT9Hw285VqASpNeae4eks5sOCH-gaJpZM4OYqbh>
> .
>

mleotta commented on README.rst:
```diff
@@ -87,107 +87,162 @@ Fletch
+VIAME_   A computer vision library designed to integrate several image and video processing algorithms together in a common              distributed processing framework, majorly targeting marine species analytics
+======= ===================================================================
+
+Code Structure and Provided Functionality
```
I was thinking about removing this section altogether or moving it different doc.   There are several directories here containing experimental content or implementation details.  I'm not sure if we want to go in this much depth in the README.  What do you think?  I'm on the fence.

aaron-bray replied:
Well, maybe you just want to highlight a subset of directories with good algorithms in there.
I just put the tree with everything in it as a starting point.
I think we can just have a subset of the tree structure with just the stuff we want to point out

aaron-bray replied:
We had a listing of vital stuff, so I thought a listing structure of arrows and sprokit would work for a high level summary of what KWIVER has from an algorithms/capabilities point of view

mleotta commented on README.rst:
```diff
@@ -100,7 +100,7 @@ To make it easier to build KWIVER, especially
 Fletch_ was developed to gather, configure and build dependent packages
 for use with KWIVER.  Fletch is a CMake_ based "super-build" that
-takes care of most of the build details for you.
+takes care of most of the built details for you.
```
typo was fixed in the wrong spot.  This one should be "build".  Now there are two typos :)

aaron-bray commented:
Ha, geeze
Fixed!

On Tue, Jul 18, 2017 at 2:36 PM, Matt Leotta <notifications@github.com>
wrote:

> *@mleotta* commented on this pull request.
> ------------------------------
>
> In README.rst
> <https://github.com/Kitware/kwiver/pull/225#discussion_r128059449>:
>
> >
> -Vital has minimal required dependencies (only Eigen_).
> -Sprokit additionally relies on Boost_.
> -Arrows and Sprokit processes are structured so that
> -the code that depends on an external package is in a directory with
> -the major dependency name (e.g. vxl, ocv). The dependencies can be
> -turned ON or OFF through CMake variables.
> +Note, This assumes your fletch was build with python support (Turn OFF if not)
>
> See other comment: fixed in the wrong spot
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Kitware/kwiver/pull/225#discussion_r128059449>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AadLH3cmZYg-mCOvdRQuCp4nbWXoU0XQks5sPPs9gaJpZM4OYqbh>
> .
>
